metaflow/plugins/kfp/s3_sensor_decorator.py

```diff
@@ -0,0 +1,43 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+
+from types import FunctionType
+from typing import Tuple
+
+def identity_formatter(path: str, flow_parameters_json: dict) -> str:
+    return key
+
+def split_s3_path(path: str) -> Tuple[str, str]:
+    path = path.replace("s3://", "")
+    bucket, key = path.split("/", 1)
+    return bucket, key
+
+class S3SensorDecorator(FlowDecorator):
+    name = 's3_sensor'
+    defaults = {
+        "path": "",
+        "timeout": 3600,
+        "polling_interval": 300,
+        "formatter": identity_formatter
```

Comment:
This is not a formatter but a callback function that returns the path?

Better to call it `"func"` else it's confusing

Reply:
`func` alone sounds too vague. How about `formatter_func` or `formatter_callback_func`?

Reply:
it's not a generic formatter, it's a function that returns the s3 path